### DISCUSSION TITLE
Is there a way to create a rule that detects that there are no new documents in an index at X time?

### DISCUSSION THREAD

**Discussion by jose-lp** [jertel/elastalert2#1181] [Q&A] [answered]
You want to receive an alert when no documents have been added to an index in x time.
In line 607 of elastalert2/elastalert.py it is possible to add this functionality and implement a new type of rule

**Comment by jertel** [comment]
Sounds like you need the Flatline rule: https://elastalert2.readthedocs.io/en/latest/ruletypes.html#flatline

**Reply by jose-lp** [reply]
I think it would not work and that I need to find a game when they find zero hits

**Reply by jertel** [reply]
If you set `threshold: 1` it will trigger when the number of events is less than 1, which is 0.

**Reply by jose-lp** [reply]
Hello @jertel
Using this setting in my rule:

`name: "No hits in sdk-regressions"

description: | 
  Rule description

limit_execution: "30-50 14 * * *"

index: "sdk-regressions"

type: "flatline"

threshold: 1

timeframe:
  minutes: 10

alert:
  - email`
  
I get this output:

`14:29:33.813Z  INFO elastalert: elastalert:  Background alerts thread 0 pending alerts sent at 2023-06-21 14:29 UTC
14:30:00.031Z  INFO elastalert: elastalert:  Queried rule No hits in sdk-regressions from 2023-06-21 14:15 UTC to 2023-06-21 14:30 UTC: 0 / 0 hits
14:30:00.054Z  INFO elastalert: elastalert:  Ran No hits in sdk-regressions from 2023-06-21 14:15 UTC to 2023-06-21 14:30 UTC: 0 query hits (0 already seen), 0 matches, 0 alerts sent
14:30:00.054Z  INFO elastalert: elastalert:  No hits in sdk-regressions range 900`

Flatline rule is not alerting when no hits. Can you see some mistake?

**Reply by jertel** [reply, marked as answer]
Hello!

You didn't wrap your rule properly with back ticks ```, so I don't know if there's a syntax mistake. I just created a brand new Elasticsearch instance with no data in it. Then I used the published ElastAlert2 Docker  image and ran the following rule. After 5 minutes it correctly alerted that "An abnormally low number of events occurred..."

```
name: flatline1
type: flatline
index: "*"
filter:
- query:
    query_string:
      query: 'blah: test'
threshold: 1
timeframe:
  minutes: 5
alert:
- debug
realert:
  minutes: 1
```

**Reply by jose-lp** [reply]
Thank you so much @jertel 

This works!
